tests-java17/test/org/immutables/tests/RecordTests.java

```diff
@@ -0,0 +1,24 @@
+/*
+   Copyright 2021 Immutables Authors and Contributors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+package org.immutables.tests;
+
+public class RecordTests {
+
+    // TODO: Test asserting that @Value.Immutable on records generates a compilation error
```

Comment:
I gave creating a test for this a serious try, but could not find an existing test asserting a compile error and could not get one to work with either the existing ProcessorRule, nor managed to create a new one.

Suggestions welcome